hh-test/utils.ts

```diff
@@ -44,27 +63,157 @@ export const utf8Encode = (payload: string) => {
+        const temp = paddForHex(BigInt(`${idAddress.slice(2, idAddress.length)}`).toString(16))
+        return hexToBytes("0x" + temp)
+    },
+    findIDAddressToBigInt: (filAddress: string) => {
+        const raw = execSync(`${PREFIX_CMD}lotus state lookup ${filAddress}"`).toString().replace("\n", "")
+        return BigInt(raw.slice(2, raw.length))
+    },
+    restart: async (params: { LOTUS_FEVM_ENABLEETHRPC: boolean }) => {
+        const exportEnvVar = `export LOTUS_FEVM_ENABLEETHRPC=${params.LOTUS_FEVM_ENABLEETHRPC}`
+        const scriptRun = `${SCRIPTS_DIR}/2_restart-localnet.sh &`
+
+        const cmd = `${PREFIX_CMD}${exportEnvVar} && ${scriptRun}"`
+
+        exec(cmd).toString().replace("\n", "")
+
+        await delay(90_000)
+    },
+    kill: () => {
+        const scriptRun = `${SCRIPTS_DIR}/4_kill-lotus-ps.sh &`
+
+        const cmd = `${PREFIX_CMD} ${scriptRun}"`
+
+        return execSync(cmd).toString().replace("\n", "")
+    },
+    registerVerifier: (filAddress: string, amount: number) => {
+        // console.log("registerVerifier", filAddress, amount)
+        const rootKey1 = _getVerifier1RootKey()
+        const cmd = `${PREFIX_CMD}lotus-shed verifreg add-verifier ${rootKey1} ${filAddress} ${amount}"`
+        return execSync(cmd).toString().replace("\n", "")
+    },
+    grantDatacap: (notaryAddress: string, filAddress: string, amount: number) => {
+        // console.log("grantDatacap", notaryAddress, filAddress, amount)
+        const cmd = `${PREFIX_CMD}lotus filplus grant-datacap --from=${notaryAddress} ${filAddress} ${amount}"`
+        return execSync(cmd).toString().replace("\n", "")
+    },
+    changeMinerOwner: (newBeneficiary: string) => {
+        // const preCmd = `${PREFIX_CMD}lotus wallet set-default ...`
+        //lotus-miner actor confirm-change-beneficiary  --really-do-it --new-beneficiary
+        const cmd = `${PREFIX_CMD}lotus-miner actor propose-change-beneficiary --really-do-it --overwrite-pending-change ${newBeneficiary} 1 1 "`
+        return execSync(cmd).toString().replace("\n", "")
+    },
+    evmInvoke: (contractFilAddress: string, payload: string) => {
+        const payloadWithout0x = payload.replace(`0x`, ``)
+        const cmd = `${PREFIX_CMD}lotus evm invoke ${contractFilAddress} ${payloadWithout0x}"`
+        return execSync(cmd).toString().replace("\n", "")
+    },
+    importDefaultWallets: () => {
+        const keyFilename = "tmp-000001.keygen"
+        try {
+            //note: try/catch because they are maybe already imported which will cause failure
+
+            writeFileSync(keyFilename, process.env.F3_PK)
+            _execute(`lotus wallet import ${keyFilename}`)
+        } catch (err) {
+            // console.log("ERR (importDefaultWallets):", err)
+        }
+        _execute(`rm -rf ${keyFilename}`)
+
+        return { fil: { address: process.env.F3_ADDR, idAddress: BigInt(process.env.F3_ID) } }
```

Comment:
F3_ADDR is not documented anywhere. Can you include it in the .env file or automatically determine the value from F3_PK? 2/2

Reply:
F3_ADDR added in the `.env.example`